analyzer/src/main/kotlin/managers/utils/MavenSupport.kt

```diff
@@ -141,7 +141,35 @@ class MavenSupport(workspaceReader: WorkspaceReader) {
                 }?.trim()
             }.toSortedSet()
 
-        private fun parseScm(scm: Scm?): VcsInfo {
+        fun getOriginalScm(mavenProject: MavenProject): Scm? {
+            // When asking Maven for the SCM URL of a POM that does not itself define an SCM URL, Maven returns the SCM
+            // URL of the first parent POM (if any) that deines one and appends the artifactIds of all child POMs to it,
+            // separated by slashes.
+            // This behavior is fundamentally broken because it invalidates the SCM URL for all VCS that cannot limit
+            // cloning to a specific path within a repository, or use a different syntax for that. Also, the assumption
+            // that the source code for a child artifact is stored in a top-level directory named like the artifactId
+            // inside the parent artifact's repository is often not correct.
+            // To address this, determine the SCM URL of the parent (if any) that is closest to the root POM and whose
+            // SCM URL still is a prefix the child POM's SCM URL.
```

Comment:
Can you move this comment to the KDoc of the function as part of this commit?

Reply:
Done.